net/pubsub.go

```diff
@@ -140,7 +140,9 @@ func (s *PubSub) watch(ctx context.Context, id thread.ID, topic *topic) {
 // subscribe to a topic for thread updates.
 func (s *PubSub) subscribe(ctx context.Context, id thread.ID, topic *topic) {
 	var err error
+	s.Lock()
 	topic.s, err = topic.t.Subscribe()
+	s.Unlock()
```

Comment:
This is necessary since `subscribe()` runs on a fired goroutine from `Add()`.
`Add()` gets the lock, adds it to the topic map and release the lock, but the goroutine of `subscribe()` writes to `topic.s` without locking it.

The problem lies on a concurrent call of `Remove()`. Even if `Remove()` is guarded, it finds the `topic` that `Add()` added, and does `topic.s.Remove()`. But The `subscribe()` call is assigning the same `topic.s` without owning it.

It might sound confusing... but this change fixes it. 

@sanderpick , maybe this assignment can be moved to the `Add` function and forget about this lock at the `subscribe()` level. But maybe it was here since `.Subscribe()` may be slow?

Reply:
Ah, good catch! Yep, it can take a bit of time to subscribe, which is why it lives here as you said.

Reply:
Great, so this sounds good to me then.